src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -572,9 +642,42 @@ - (MTRDeviceController * _Nullable)_startDeviceController:(MTRDeviceControllerSt
         return nil;
     }
 
+    id<MTRDeviceControllerStorageDelegate> storageDelegate;
+    dispatch_queue_t storageDelegateQueue;
+    NSUUID * uniqueIdentifier;
+    if ([startupParams isKindOfClass:[MTRDeviceControllerStartupParameters class]]) {
+        MTRDeviceControllerStartupParameters * params = startupParams;
+        storageDelegate = params.storageDelegate;
+        storageDelegateQueue = params.storageDelegateQueue;
+        uniqueIdentifier = params.uniqueIdentifier;
+    } else {
```

Comment:
does it make sense to have an - else if ([startupParams isKindOfClass:[MTRDeviceControllerStartupParams class]])  and return nil in the else case.

Reply:
This is an internal method.... but yes, we could do that just to be safe. Done.